src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -75,12 +86,72 @@ void SensorFusion::updateWorld(
+    // https://github.com/UBC-Thunderbots/Software/issues/960
+    double min_valid_x = Util::DynamicParameters->getAIControlConfig()
+                             ->getRefboxConfig()
+                             ->MinValidX()
+                             ->value();
+    double max_valid_x = Util::DynamicParameters->getAIControlConfig()
+                             ->getRefboxConfig()
+                             ->MaxValidX()
+                             ->value();
+    bool ignore_invalid_camera_data = Util::DynamicParameters->getAIControlConfig()
+                                          ->getRefboxConfig()
+                                          ->IgnoreInvalidCameraData()
+                                          ->value();
+    ;
+
+
+
+    SSL_DetectionFrame detection_frame = ssl_detection_frame;
+    // We invert the field side if we explicitly choose to override the values
+    // provided by refbox. The 'defending_positive_side' parameter dictates the side
+    // we are defending if we are overriding the value
+    // TODO remove as part of https://github.com/UBC-Thunderbots/Software/issues/960
+    if (Util::DynamicParameters->getAIControlConfig()
+            ->getRefboxConfig()
+            ->OverrideRefboxDefendingSide()
+            ->value() &&
+        Util::DynamicParameters->getAIControlConfig()
+            ->getRefboxConfig()
+            ->DefendingPositiveSide()
+            ->value())
+    {
+        invertFieldSide(detection_frame);
+    }
+
+    std::optional<TimestampedBallState> new_ball_state;
+    if (isCameraEnabled(detection_frame))
+    {
+        new_ball_state = createTimestampedBallState(createBallDetections(
+            {detection_frame}, min_valid_x, max_valid_x, ignore_invalid_camera_data));
+
+        // TODO remove Util::DynamicParameters as part of
+        // https://github.com/UBC-Thunderbots/Software/issues/960
+        if (Util::DynamicParameters->getAIControlConfig()
+                ->getRefboxConfig()
+                ->FriendlyColorYellow()
+                ->value())
+        {
+            friendly_team = createFriendlyTeam(
+                createTeamDetection({detection_frame}, TeamColour::YELLOW, min_valid_x,
+                                    max_valid_x, ignore_invalid_camera_data));
+            enemy_team = createEnemyTeam(
+                createTeamDetection({detection_frame}, TeamColour::BLUE, min_valid_x,
+                                    max_valid_x, ignore_invalid_camera_data));
+        }
+        else
+        {
+            friendly_team = createFriendlyTeam(
+                createTeamDetection({detection_frame}, TeamColour::BLUE, min_valid_x,
+                                    max_valid_x, ignore_invalid_camera_data));
+            enemy_team = createEnemyTeam(
+                createTeamDetection({detection_frame}, TeamColour::YELLOW, min_valid_x,
+                                    max_valid_x, ignore_invalid_camera_data));
+        }
```

Comment:
It seems like there's a fair amount of duplication here, perhaps we could create a `friendly_team_color` and an `enemy_team_color` variable and then just using that in the calls to`createTeamDetection`?

Reply:
Yeah I can just make blue_team and yellow_team variables and pass them in.

Reply:
done